### PR TITLE
bug 1391079 - update precomplete file after widevine signing

### PR DESCRIPTION
**Pull request by escapewindow**
@JohanLorenzo Are you up for this one? @catlee also has a lot of background here.

### CODE REVIEW

**Comment by escapewindow** [comment]
Testing is at https://tools.taskcluster.net/groups/O-913QLeTvSj6Uy4kY0luQ . Mac looks great. Linux seems to have an issue - I may need a followup fix. I suspect it's trying to sign the `firefox` directory... I may need to just `os.path.isfile` in the right place.

File: signingscript/sign.py [hunk 1]
```diff
@@ -422,13 +428,68 @@ def _get_widevine_signing_files(file_list):
...
+    with open(path, "r") as fh:
+        after = fh.readlines()
+    # Create diff file
+    diff_path = os.path.join(context.config['work_dir'], "precomplete.diff")
+    with open(diff_path, "w") as fh:
+        for line in difflib.ndiff(before, after):
+            fh.write(line)
+    utils.copy_to_dir(
+        diff_path, context.config['artifact_dir'],
+        target="public/logs/precomplete.diff"
```
Comment (JohanLorenzo) [inline signingscript/sign.py]: Minor issue: The file doesn't display in the browser. We should fix this in https://github.com/mozilla-releng/scriptworker/blob/65b110f21fed199268062191ec9cb34a2bf878ea/scriptworker/artifacts.py#L27

File: signingscript/sign.py [hunk 2]
```diff
@@ -350,31 +352,35 @@ def build_signtool_cmd(context, from_, fmt, to=None):
...
             log.debug("Adding {} to the sigfile paths...".format(to))
             makedirs(os.path.dirname(to))
             tasks.append(asyncio.ensure_future(sign_file(
                 context, from_, fmt, to=to
             )))
-            all_files.append("{}".format(to))
+            all_files.append(to)
         await raise_future_exceptions(tasks)
-        # Append sig_files to the archive
+        remove_extra_files(tmp_dir, all_files)
```
Comment (escapewindow) [inline signingscript/sign.py]: Anything outside of all_files. When i looked at the logs, I saw lines like `copying path/to/file.sig.tmp to path/to/file.sig` and i was worried that without cleanup, we might have a lot of cruft in `precomplete`. This turns out to not be the case. However, now that I have the code, I think it's a good preventative measure.

File: signingscript/test/test_sign.py [hunk 3]
```diff
@@ -322,6 +326,51 @@ def test_get_widevine_signing_files(filenames, expected):
...
+    all_files = []
+    for f in extra + good:
+        path = os.path.join(work_dir, f)
+        makedirs(os.path.dirname(path))
+        with open(path, "w") as fh:
+            fh.write("x")
+        if f in good:
+            all_files.append(path)
+    for f in good:
+        assert os.path.exists(os.path.join(work_dir, f))
```
Comment (escapewindow) [inline signingscript/test/test_sign.py]: We assert that they don't exist lower in the test.

File: signingscript/sign.py [hunk 4]
```diff
@@ -422,13 +428,68 @@ def _get_widevine_signing_files(file_list):
...
+
+    """
+    all_files = [os.path.realpath(f) for f in glob.glob(os.path.join(top_dir, '**', '*'), recursive=True)]
+    good_files = [os.path.realpath(f) for f in file_list]
+    extra_files = list(set(all_files) - set(good_files))
+    for f in extra_files:
+        if os.path.isfile(f):
+            log.warning("Extra file to clean up: {}".format(f))
+            rm(f)
+    return extra_files
```
Comment (escapewindow) [inline signingscript/sign.py]: I used to. Just added to the test :)

File: signingscript/sign.py (hunk 1 above)
Comment (escapewindow) [inline signingscript/sign.py]: +1, I noticed this too.

**Comment by escapewindow** [comment]
Thank you! I've been sending a ton of reviews your way, and you're always helpful.

**Pull request merged**